src/uct/ib/ud/base/ud_iface.c

```diff
@@ -199,6 +200,22 @@ uct_ud_iface_create_qp(uct_ud_iface_t *self, const uct_ud_iface_config_t *config
     qp_init_attr.cap.max_recv_sge    = 1;
     qp_init_attr.cap.max_inline_data = config->super.tx.min_inline;
 
+    if ((dev->max_inline_data > 0) &&
+        (qp_init_attr.cap.max_inline_data > dev->max_inline_data)) {
+        ucs_diag("create QP: TX inline: using device cap %uB (configured %uB)",
+                 dev->max_inline_data, qp_init_attr.cap.max_inline_data);
+
+        qp_init_attr.cap.max_inline_data = dev->max_inline_data;
+    }
+
+    if ((dev->max_sq_sge > 0) &&
+        (qp_init_attr.cap.max_send_sge > dev->max_sq_sge)) {
```

Comment:
```suggestion
    if (qp_init_attr.cap.max_send_sge > dev->max_sq_sge) {
```

Reply:
it is needed because currently only EFA populates that value, else it is set to zero.

Reply:
fixed by using `UINT16_MAX` default value